doc/on-behalf-of.md

```diff
@@ -0,0 +1,50 @@
+# On-Behalf-Of Authorization
+
+The feature can be enabled with the checkbox from the plugin configuration screen.
+⚠️ When provisioning, you also need to set `oauthPassThru` to `true`, such as the following example has done.
+
+```yaml
+apiVersion: 1
+datasources:
+  - name: Azure Data Explorer
+    type: grafana-azure-data-explorer-datasource
+    access: proxy
+    basicAuth: false
+    editable: true
+    jsonData:
+      onBehalfOf: true
+      oauthPassThru: true
+      clientId: <your client UUID>
+      clusterUrl: <your cluster URL>
+      tenantId: <your tenant UUID>
+      tlsAuth: false
+      tlsAuthWithCACert: false
+      defaultDatabase: <your default database>
+    secureJsonData:
+      clientSecret: <your client secret>
+      tlsCACert: ''
+      tlsClientCert: ''
+      tlsClientKey: ''
+    version: 1
+```
+
+⚠️ Only compatible with Grafana 8.3.4 or later above.
+⚠️ Grafana alert-rules do not work with on-behalf-of authorization.
+
+
+## Setup
+
+Configure Grafana to use OAuth2 with Azure Active Directory as [documented](https://grafana.com/docs/grafana/latest/auth/azuread/). The `[auth.azuread]` `scopes` (or `$GF_AUTH_AZUREAD_SCOPES`) must contain “openid email profile”.
+
+ID tokens must be enabled with a checkbox found on the [Azure portal](https://portal.azure.com/) under “App Registrations” → the respective application → “Manage” → “Authentication”.
+
+In addition to the “Microsoft Graph” `User.Read`, a special “Azure Data Explorer” `user_impersonation` permission must be enabled on the [Azure portal](https://portal.azure.com/)  under “App Registrations” → the respective application → “Manage” → “API permissions”. ⚠️ “Admin consent” is required.
+
```

Comment:
Please rewrite this to a _step by step_ setup. E.g:
1. Configure Grafana to use OAuth2...
2. Enable ID tokens.
3. Allow user impersonation permission
4. Enable "Admin consent" under “App Registrations” → the respective application → “Security” → “Permissions”.

Since the admin consent part requires an actual Azure admin to make changes in the account I think you should elaborate a bit on this. You can add something like this _Enabling "Admin consent" will grant consent on behalf of all users in the current tenant, ensuring the users will not be required to consent when using the application._

Reply:
Admin consent is enabled by default already, but OK. Microsoft has plenty of documentation on how to configure the consent. What we need to provide is a checklist (as provided here).

Reply:
It should be clear to the user why admin consent is necessary. As said before, add this text:
_Enabling "Admin consent" will grant consent on behalf of all users in the current tenant, ensuring the users will not be required to consent when using the application._